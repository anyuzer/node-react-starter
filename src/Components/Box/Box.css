.box{
    position: relative;
    box-sizing: border-box;
    width: 400px;
    height: 200px;
    border: 2px solid navy;
    border-radius: 4px;
    text-align:center;
    line-height: 200px;
}