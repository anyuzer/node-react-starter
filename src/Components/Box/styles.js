export default {
    box: {
        position: "relative",
        boxSizing: "borderBox",
        width: "400px",
        height: "200px",
        border: "2px solid navy",
        borderRadius: "4px",
        textAlign: "center",
        lineHeight: "200px"
    }
};