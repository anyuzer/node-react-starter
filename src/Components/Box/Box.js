import React from 'react';
import styles from './styles';

class Box extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      message: "Hello World"
    };
  }

  updateMessage() {
    this.setState({
      message: (this.state.message === 'Hello World' ? 'Goodbye World' : 'Hello World')
    });
  }

  render() {
    return (
      <div style={styles.box} onClick={this.updateMessage.bind(this)}>
        {this.state.message}
      </div>
    );
  }
}

export default Box;