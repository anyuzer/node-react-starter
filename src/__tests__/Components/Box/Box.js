import renderer from 'react-test-renderer';
import React from 'react';
import Box from '../../../Components/Box/Box';

describe('Box Component', () => {
  const component = renderer.create(<Box />);
  it('should match the Box component snapshot', () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.onClick();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.onClick();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});