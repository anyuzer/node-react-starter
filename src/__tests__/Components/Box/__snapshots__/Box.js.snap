// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Component should match the Box component snapshot 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "border": "2px solid navy",
      "borderRadius": "4px",
      "boxSizing": "borderBox",
      "height": "200px",
      "lineHeight": "200px",
      "position": "relative",
      "textAlign": "center",
      "width": "400px",
    }
  }>
  Hello World
</div>
`;

exports[`Box Component should match the Box component snapshot 2`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "border": "2px solid navy",
      "borderRadius": "4px",
      "boxSizing": "borderBox",
      "height": "200px",
      "lineHeight": "200px",
      "position": "relative",
      "textAlign": "center",
      "width": "400px",
    }
  }>
  Goodbye World
</div>
`;

exports[`Box Component should match the Box component snapshot 3`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "border": "2px solid navy",
      "borderRadius": "4px",
      "boxSizing": "borderBox",
      "height": "200px",
      "lineHeight": "200px",
      "position": "relative",
      "textAlign": "center",
      "width": "400px",
    }
  }>
  Hello World
</div>
`;
